discord/sink.py

```diff
@@ -0,0 +1,775 @@
+        the sender changes its SSRC identifier.
+    report_blocks: Sequence[:class:`RTCPReceiverReport`]
+        Sequence of :class:`RTCPReceiverReport` objects that tell statistics.
+        Receivers do not carry over statistics when a source changes its SSRC
+        identifier due to a collision.
+    extension: :class:`bytes`
+        Profile-specific extension that may or may not contain a value.
+    """
+    __slots__ = RTCPPacket.__slots__ + (
+        "ssrc", "nts", "rts", "spc", "soc", "report_blocks",
+        "extension"
+    )
+
+    def __init__(self, version_flag: int, rtcp_type: RTCPMessageType, length: int, data: bytes):
+        super().__init__(version_flag, rtcp_type, length)
+
+        fmt = ">IQ3I"
+        buf_size = struct.calcsize(fmt)
+        self.ssrc, self.nts, self.rts, self.spc, self.soc = \
+            struct.unpack(fmt, data[:buf_size])
+
+        self._parse_report_and_extension(data[buf_size:])
+
+
+class RTCPReceiverReportPacket(RTCPPacket):
+    """RTCP Receiver Report packet which provides quality feedback.
+
+    Read in detail here: https://www.freesoft.org/CIE/RFC/1889/20.htm
+
+    Extends :class:`RTCPPacket` and inherits its attributes.
+
+    Attributes
+    ----------
+    ssrc: :class:`int`
+        The synchronization source identifier for the originator of this SR packet.
+    report_blocks: Sequence[:class:`RTCPReceiverReport`]
+        Sequence of :class:`RTCPReceiverReport` objects that tell statistics.
+        Receivers do not carry over statistics when a source changes its SSRC
+        identifier due to a collision.
+    extension: :class:`bytes`
+        Profile-specific extension that may or may not contain a value.
+    """
+    __slots__ = RTCPPacket.__slots__ + (
+        "ssrc", "report_blocks", "extension"
+    )
+
+    def __init__(self, version_flag: int, rtcp_type: RTCPMessageType, length: int, data: bytes):
+        super().__init__(version_flag, rtcp_type, length)
+
+        fmt = ">I"
+        buf_size = struct.calcsize(fmt)
+        self.ssrc = struct.unpack(fmt, data[:buf_size])[0]
+        self._parse_report_and_extension(data[buf_size:])
+
+
+class RTCPSourceDescriptionPacket(RTCPPacket):
+    """Source Description packet which describes sources.
+
+    Read in detail here: https://www.freesoft.org/CIE/RFC/1889/23.htm
+
+    Extends :class:`RTCPPacket` and inherits its attributes.
+
+    Attributes
+    ----------
+    chunks: Sequence[:class:`RTCPSourceDescriptionChunk`]
+        Sequence of chunks that contain items.
+    """
+    __slots__ = RTCPPacket.__slots__ + ("chunks",)
+
+    def __init__(self, version_flag: int, rtcp_type: RTCPMessageType, length: int, data: bytes):
+        super().__init__(version_flag, rtcp_type, length)
+
+        self.chunks = []
+        for _ in range(self.rc):
+            chunk, offset = self._parse_chunk(data)
+            data = data[offset:]
+            self.chunks.append(chunk)
+
+    def _parse_chunk(self, data):
```

Comment:
I understand it's a private method so feel free to ignore, but shouldn't this still have a TypedDict data payload type?

Reply:
Not sure I understand what you mean. If this function were typed, would it not be `_parse_chunk(self, data: bytes) -> Tuple[RTCPSourceDescriptionChunk, int]`? I can add the typing since private methods elsewhere in the project also have typing.

Reply:
Perfect.